src/lib/ScannerRunCommand.ts

```diff
@@ -1,24 +1,72 @@
+import {flags} from '@salesforce/command';
 import {Messages, SfdxError} from '@salesforce/core';
 import {AnyJson} from '@salesforce/ts-types';
 import {ScannerCommand} from './ScannerCommand';
-import {RecombinedRuleResults} from '../types';
+import {RecombinedRuleResults, SfgeConfig} from '../types';
 import {RunOutputProcessor} from './util/RunOutputProcessor';
 import {Controller} from '../Controller';
+import {CUSTOM_CONFIG} from '../Constants';
 import {OUTPUT_FORMAT, RunOptions} from './RuleManager';
+import {FileHandler} from './util/FileHandler';
 import untildify = require('untildify');
+import globby = require('globby');
+import path = require('path');
 import normalize = require('normalize-path');
 
 // Initialize Messages with the current plugin directory
 Messages.importMessagesDirectory(__dirname);
 
 // Load the specific messages for this file. Messages from @salesforce/command, @salesforce/core,
 // or any library that is using the messages framework can also be loaded this way.
-const messages = Messages.loadMessages('@salesforce/sfdx-scanner', 'run');
+const messages = Messages.loadMessages('@salesforce/sfdx-scanner', 'run-common');
 // This code is used for internal errors.
 export const INTERNAL_ERROR_CODE = 1;
 
 export abstract class ScannerRunCommand extends ScannerCommand {
 
+	/**
+	 * There are flags that are common to all variants of the run command. We can define those flags
+	 * here to avoid duplicate code.
+	 * @protected
+	 */
+	protected static flagsConfig = {
+		verbose: flags.builtin(),
+		// BEGIN: Flags related to results processing.
+		format: flags.enum({
+			char: 'f',
+			description: messages.getMessage('flags.formatDescription'),
+			longDescription: messages.getMessage('flags.formatDescriptionLong'),
+			options: [OUTPUT_FORMAT.CSV, OUTPUT_FORMAT.HTML, OUTPUT_FORMAT.JSON, OUTPUT_FORMAT.JUNIT, OUTPUT_FORMAT.SARIF, OUTPUT_FORMAT.TABLE, OUTPUT_FORMAT.XML]
+		}),
+		outfile: flags.string({
+			char: 'o',
+			description: messages.getMessage('flags.outfileDescription'),
+			longDescription: messages.getMessage('flags.outfileDescriptionLong')
+		}),
+		'severity-threshold': flags.integer({
+			char: 's',
+			description: messages.getMessage('flags.sevthresholdDescription'),
+			longDescription: messages.getMessage('flags.sevthresholdDescriptionLong'),
+			exclusive: ['json'],
+			min: 1,
+			max: 3
+		}),
+		'normalize-severity': flags.boolean({
+			description: messages.getMessage('flags.normalizesevDescription'),
+			longDescription: messages.getMessage('flags.normalizesevDescriptionLong')
+		}),
+		// END: Flags related to results processing.
+		// BEGIN: Flags related to targeting.
+		projectdir: flags.array({
```

Comment:
IMPORTANT CHANGE: To allow `--projectdir/-p` to be used by both commands, it has been made optional. As such, `scanner:run:dfa`'s flag validation checks that it's present.

Reply:
I can see your rationale. Thinking about this a little more, wouldn't the description need to be a little different? For `scanner:run`, we should mention that the flag is specific to `--engine sfge`.
I'm fine with how you've made this a common flag for both commands. We could probably compare how the documentation renders to see if we should handle them differently after all.